### PR TITLE
Document RISC-V specific directives

### PR DESCRIPTION
**Pull request by palmer-dabbelt**

### CODE REVIEW

**Comment by jim-wilson** [comment]
I added this patch to the FSF Binutils tree, with some typos fixed, and some misc cleanups included.

File: gas/doc/c-riscv.texi [hunk 1]
```diff
@@ -47,3 +47,89 @@ the soft-float calling convention.
...
+@cindex Option directive
+@item .option @var{argument}
+Modifies RISC-V specific assembler options in-line with the assembly code.
+This is used when particular instruction sequences must be assembler with a
+specific set of options.  For example, since we relax addressing sequences to
+shorter GP-relative sequences when possible the initial load of GP must not be
+relaxed and should be emitted as something like
+
+.option push
+.option pop
```
Comment (ilg-ul) [inline gas/doc/c-riscv.texi]: is this first `pop` correct?

Comment (jim-wilson) [reply gas/doc/c-riscv.texi]: Fixed in the patch I commited.  See
https://sourceware.org/ml/binutils/2017-12/msg00021.html

**Comment by palmer-dabbelt** [comment]
I'm closing this because it was merged.

**Pull request closed without merging**